docs/architecture.md

```diff
@@ -0,0 +1,218 @@
+
+Skenario asks dispatcher:
+    - Dispatcher, tell me how many pods do I need?
+The dispatcher gets a request and delegates it to the HPA plugin:
+    - The HPA plugin, tell me how many pods does Skenario need?
+The HPA plugin gets a request and delegates it to Kubernetes:     
+    - Kubernetes, tell me how many pods does Skenario need?
+Kubernetes gets a request and gives a response, the number of pods, to the HPA plugin:
+    - The HPA plugin, here is the number of pods that Skenario needs.
+The HPA plugin gets a response and passes it to the dispatcher:
+    - Dispatcher, here is the number of pods that Skenario needs.
+The dispatcher gets a response and passes it to Skenario:
+    - Skenario, here is the number of pods that you need.
+
+From an architecture perspective, the diagram below reflects the scaling lifecycle.
+                                
+```
+
+Skenario ---> Dispatcher ---> HPA plugin ---> Kubernetes
+
+Skenario <--- Dispatcher <--- HPA plugin <--- Kubernetes
+
+```
+
+### HPA plugin
+
+plugin-k8s is the HPA plugin. The HPA plugin creates a HorizontalController 
+for each Simulation Environment. It keeps track of pods and stats per environment 
+and provides them to the controller via mocks and fakes, injected at construction.
+
+Instead of starting the controller (which runs goroutines) the plugin drives 
+the controller by calling the reconcileAutoscaler method with the current HPA object. 
+The controller updates the HPA object through a reactor on the fake client.
+ 
+## Vertical scaling in Skenario
+
+For more information how Skenario scales vertically see [concepts.md].
+Vertical scaling looks like communication between components. 
+
+Skenario asks dispatcher:
+    - Dispatcher, tell me which size of pods do I need?
+The dispatcher gets a request and delegates it to the VPA plugin:
+    - The VPA plugin, tell me which size of pods does Skenario need?
+The VPA plugin gets a request and delegates it to Kubernetes:     
+    - Kubernetes, tell me which size of pods does Skenario need?
+Kubernetes gets a request and gives a response, the size (cpu capacity) that is
+appropriate for every pod, to the VPA plugin:
+    - The VPA plugin, here is the size which is appropriate for every pod in Skenario.
+The HPA plugin gets a response and passes it to the dispatcher:
+    - Dispatcher, here is the size which is appropriate for every pod in Skenario.
+The dispatcher gets a response and passes it to Skenario:
+    - Skenario, here is the size which is appropriate for every pod in Skenario.
+
+From an architecture perspective, the diagram below reflects the scaling lifecycle.
+                                
+```
+
+Skenario ---> Dispatcher ---> VPA plugin ---> Kubernetes
+
+Skenario <--- Dispatcher <--- VPA plugin <--- Kubernetes
+
+```
+
+### VPA plugin
+
+plugin-k8s-vpa is the VPA plugin. As vertical scaling is more complicated 
+than horizontal we are interested only in the recommender component and 
```

Comment:
You might mention which other parts exist (Updater and Admission Controller) and briefly how they are handled in Skenario (rather than the plugin).

Reply:
Good point! I added it.